another cftime failure

```
________ TestNetCDF3ViaNetCDF4Data.test_roundtrip_cftime_datetime_data _________

dates = array([cftime.DatetimeProlepticGregorian(0001-01-01 00:00:00),
       cftime.DatetimeProlepticGregorian(0001-01-02 00:00:00)],
      dtype=object)
units = 'days since 0001-01-01 00:00:00.000000'
calendar = 'proleptic_gregorian'

    def encode_cf_datetime(dates, units=None, calendar=None):
        """Given an array of datetime objects, returns the tuple `(num, units,
        calendar)` suitable for a CF compliant time variable.
    
        Unlike `date2num`, this function can handle datetime64 arrays.
    
        See also
        --------
        cftime.date2num
        """
        dates = np.asarray(dates)
    
        if units is None:
            units = infer_datetime_units(dates)
        else:
            units = _cleanup_netcdf_time_units(units)
    
        if calendar is None:
            calendar = infer_calendar_name(dates)
    
        delta, ref_date = _unpack_netcdf_time_units(units)
        try:
            if calendar not in _STANDARD_CALENDARS or dates.dtype.kind == "O":
                # parse with cftime instead
>               raise OutOfBoundsDatetime
E               pandas._libs.tslibs.np_datetime.OutOfBoundsDatetime

xarray/coding/times.py:400: OutOfBoundsDatetime

During handling of the above exception, another exception occurred:

self = <xarray.tests.test_backends.TestNetCDF3ViaNetCDF4Data object at 0x7fd54f4cb780>

    @requires_cftime
    def test_roundtrip_cftime_datetime_data(self):
        from .test_coding_times import _all_cftime_date_types
    
        date_types = _all_cftime_date_types()
        for date_type in date_types.values():
            times = [date_type(1, 1, 1), date_type(1, 1, 2)]
            expected = Dataset({"t": ("t", times), "t0": times[0]})
            kwargs = {"encoding": {"t0": {"units": "days since 0001-01-01"}}}
            expected_decoded_t = np.array(times)
            expected_decoded_t0 = np.array([date_type(1, 1, 1)])
            expected_calendar = times[0].calendar
    
            with warnings.catch_warnings():
                if expected_calendar in {"proleptic_gregorian", "gregorian"}:
                    warnings.filterwarnings("ignore", "Unable to decode time axis")
    
>               with self.roundtrip(expected, save_kwargs=kwargs) as actual:

xarray/tests/test_backends.py:483: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/share/miniconda/envs/xarray-tests/lib/python3.6/contextlib.py:81: in __enter__
    return next(self.gen)
xarray/tests/test_backends.py:243: in roundtrip
    self.save(data, path, **save_kwargs)
xarray/tests/test_backends.py:265: in save
    path, engine=self.engine, format=self.file_format, **kwargs
xarray/core/dataset.py:1536: in to_netcdf
    invalid_netcdf=invalid_netcdf,
xarray/backends/api.py:1071: in to_netcdf
    dataset, store, writer, encoding=encoding, unlimited_dims=unlimited_dims
xarray/backends/api.py:1117: in dump_to_store
    store.store(variables, attrs, check_encoding, writer, unlimited_dims=unlimited_dims)
xarray/backends/common.py:293: in store
    variables, attributes = self.encode(variables, attributes)
xarray/backends/common.py:382: in encode
    variables, attributes = cf_encoder(variables, attributes)
xarray/conventions.py:748: in cf_encoder
    new_vars = {k: encode_cf_variable(v, name=k) for k, v in variables.items()}
xarray/conventions.py:748: in <dictcomp>
    new_vars = {k: encode_cf_variable(v, name=k) for k, v in variables.items()}
xarray/conventions.py:250: in encode_cf_variable
    var = coder.encode(var, name=name)
xarray/coding/times.py:446: in encode
    data, encoding.pop("units", None), encoding.pop("calendar", None)
xarray/coding/times.py:419: in encode_cf_datetime
    num = _encode_datetime_with_cftime(dates, units, calendar)
xarray/coding/times.py:366: in _encode_datetime_with_cftime
    return np.vectorize(encode_datetime)(dates)
/usr/share/miniconda/envs/xarray-tests/lib/python3.6/site-packages/numpy/lib/function_base.py:2755: in __call__
    return self._vectorize_call(func=func, args=vargs)
/usr/share/miniconda/envs/xarray-tests/lib/python3.6/site-packages/numpy/lib/function_base.py:2825: in _vectorize_call
    ufunc, otypes = self._get_ufunc_and_otypes(func=func, args=args)
/usr/share/miniconda/envs/xarray-tests/lib/python3.6/site-packages/numpy/lib/function_base.py:2785: in _get_ufunc_and_otypes
    outputs = func(*inputs)
xarray/coding/times.py:364: in encode_datetime
    return np.nan if d is None else cftime.date2num(d, units, calendar)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   AttributeError: 'cftime._cftime.DatetimeProlepticGregorian' object has no attribute 'tzinfo'

cftime/_cftime.pyx:186: AttributeError
```

--- a/doc/whats-new.rst
+++ b/doc/whats-new.rst
@@ -36,6 +36,10 @@ Bug fixes
 - Fix regression introduced in v0.14.0 that would cause a crash if dask is installed
   but cloudpickle isn't (:issue:`3401`) by `Rhys Doyle <https://github.com/rdoyle45>`_
 
+- Sync with cftime by removing `dayofwk=-1` for cftime>=1.0.4. 
+   By `Anderson Banihirwe <https://github.com/andersy005>`_.
+
+
 Documentation
 ~~~~~~~~~~~~~
 

--- a/xarray/coding/cftime_offsets.py
+++ b/xarray/coding/cftime_offsets.py
@@ -237,7 +237,14 @@ def _shift_month(date, months, day_option="start"):
         day = _days_in_month(reference)
     else:
         raise ValueError(day_option)
-    return date.replace(year=year, month=month, day=day)
+    if LooseVersion(cftime.__version__) < LooseVersion("1.0.4"):
+        # dayofwk=-1 is required to update the dayofwk and dayofyr attributes of
+        # the returned date object in versions of cftime between 1.0.2 and
+        # 1.0.3.4.  It can be removed for versions of cftime greater than
+        # 1.0.3.4.
+        return date.replace(year=year, month=month, day=day, dayofwk=-1)
+    else:
+        return date.replace(year=year, month=month, day=day)
 
 
 def roll_qtrday(other, n, month, day_option, modby=3):

--- a/xarray/coding/cftime_offsets.py
+++ b/xarray/coding/cftime_offsets.py
@@ -50,6 +50,7 @@ import numpy as np
 from ..core.pdcompat import count_not_none
 from .cftimeindex import CFTimeIndex, _parse_iso8601_with_reso
 from .times import format_cftime_datetime
+from distutils.version import LooseVersion
 
 
 def get_date_type(calendar):
@@ -222,6 +223,8 @@ def _adjust_n_years(other, n, month, reference_day):
 def _shift_month(date, months, day_option="start"):
     """Shift the date to a month start or end a given number of months away.
     """
+    import cftime
+
     delta_year = (date.month + months) // 12
     month = (date.month + months) % 12
 

--- a/xarray/coding/cftime_offsets.py
+++ b/xarray/coding/cftime_offsets.py
@@ -237,11 +237,7 @@ def _shift_month(date, months, day_option="start"):
         day = _days_in_month(reference)
     else:
         raise ValueError(day_option)
-    # dayofwk=-1 is required to update the dayofwk and dayofyr attributes of
-    # the returned date object in versions of cftime between 1.0.2 and
-    # 1.0.3.4.  It can be removed for versions of cftime greater than
-    # 1.0.3.4.
-    return date.replace(year=year, month=month, day=day, dayofwk=-1)
+    return date.replace(year=year, month=month, day=day)
 
 
 def roll_qtrday(other, n, month, day_option, modby=3):

--- a/xarray/coding/cftimeindex.py
+++ b/xarray/coding/cftimeindex.py
@@ -107,7 +107,12 @@ def _parse_iso8601_with_reso(date_type, timestr):
             # TODO: Consider adding support for sub-second resolution?
             replace[attr] = int(value)
             resolution = attr
-
+    if LooseVersion(cftime.__version__) < LooseVersion("1.0.4"):
+        # dayofwk=-1 is required to update the dayofwk and dayofyr attributes of
+        # the returned date object in versions of cftime between 1.0.2 and
+        # 1.0.3.4.  It can be removed for versions of cftime greater than
+        # 1.0.3.4.
+        replace["dayofwk"] = -1
     return default.replace(**replace), resolution
 
 

--- a/xarray/coding/cftimeindex.py
+++ b/xarray/coding/cftimeindex.py
@@ -96,6 +96,8 @@ def parse_iso8601(datetime_string):
 
 
 def _parse_iso8601_with_reso(date_type, timestr):
+    import cftime
+
     default = date_type(1, 1, 1)
     result = parse_iso8601(timestr)
     replace = {}

--- a/xarray/coding/cftimeindex.py
+++ b/xarray/coding/cftimeindex.py
@@ -108,11 +108,6 @@ def _parse_iso8601_with_reso(date_type, timestr):
             replace[attr] = int(value)
             resolution = attr
 
-    # dayofwk=-1 is required to update the dayofwk and dayofyr attributes of
-    # the returned date object in versions of cftime between 1.0.2 and
-    # 1.0.3.4.  It can be removed for versions of cftime greater than
-    # 1.0.3.4.
-    replace["dayofwk"] = -1
     return default.replace(**replace), resolution
 
 

